src/dbnode/storage/shard.go

```diff
@@ -816,10 +816,16 @@ func (s *dbShard) writeAndIndex(
 		commitLogSeriesID          ident.ID
 		commitLogSeriesTags        ident.Tags
 		commitLogSeriesUniqueIndex uint64
+		shouldWrite                = true
 	)
 	if writable {
 		// Perform write
-		err = entry.Series.Write(ctx, timestamp, value, unit, annotation)
+		shouldWrite, err = entry.Series.Write(ctx, timestamp, value, unit, annotation)
+		if err != nil {
```

Comment:
Any reason you moved the error handling code around? it looks like the old code only had to call `DecrementReaderWriterCount` once but this one has to do it multiple times

Reply:
So this should still call `DecrementReaderCount` once, depending on which operations error; the previous code would still do these calls:

```
commitLogSeriesID = entry.Series.ID()			commitLogSeriesID = entry.Series.ID()
commitLogSeriesTags = entry.Series.Tags()			commitLogSeriesTags = entry.Series.Tags()
commitLogSeriesUniqueIndex = entry.Index
```

for no reason if `.Write(..)` errored. Happy to revert it though if this is fine

Reply:
yeah I think i'd prefer not to change this for now as I think its a little more brittle now (cleanup code more spread out)